app/models/student.rb

```diff
@@ -8,6 +8,7 @@ class Student < ActiveRecord::Base
 
   belongs_to :homeroom, optional: true, counter_cache: true
   belongs_to :school
+  has_one :student_photo
```

Comment:
I saw some earlier comments about the keys here, so not sure if this is entirely relevant, but this looks to me like the storer code might create a `StudentPhoto` record for each photo as it is imported.  If that's still the case, then I'm not sure how `has_one` works if there are multiple records on the other side of the association (I think `has_one` just adds `LIMIT 1` to the query and nothing smarter).

This also doesn't matter at all for the initial import, just seems like a place a sneaky bug might come up.

Reply:
Ooooh good catch here. Do you feel strongly about `has_one` versus `has_many`? We could either store the most recent photo for each student in AWS and our database, or store all the photos and select the most recent one in application code.

You're right to point out that the import code looks like it's currently configured for `has_many`. The `"student / upload date / image"` file structure implies multiple images per student and we don't overwrite anywhere. 

I think the `has_one` approach is better because it means we store fewer photos. That means cheaper AWS bills and less clutter lying around.

My instinct is to change this so that the import code keeps us strictly `has_one` (and maybe write some stronger validation on the model), but if you want to offer any reasons why `has_many` is the better choice, I'd really like to hear because I can't think of any myself off the top of my head.

Reply:
@alexsoble I'm happy to go with whatever you decide!  I think it's good to just :ship this since we're not setting up a recurring job yet anyway (on our end or on John's).

Initially, I was thinking that writing them all would be simpler.  We don't have to track changes over time - we just get photos, and we write them to S3, there's no other moving parts.  Nothing can change over time, except for which one a given student points to.  And we'll have the information in the database and logs to understand or debug that if it seems weird.

For costs, I'm assuming that any S3 storage costs here will be almost nothing, and if it does cost anything it'll cost less than an hour of our time each month, so I wasn't thinking it was super meaningful to optimize.

So what I was initially thinking was sort of a hybrid of the two options you described: we store everything, and we make it easy to query for the current photo.  Maybe another iteration on that might be making a `latest_student_photo` method on the `student` model?  That would remove the second key that you were wondering about, although it would require adding an index in its place to make it fast to query for the latest photo.